roles/calico/templates/calico.yml.j2

```diff
@@ -210,9 +210,11 @@ spec:
                 configMapKeyRef:
                   name: calico-config
                   key: etcd_cert
-            # Auto-detect the BGP IP address.
+            # Use hostIP as the BGP IP address.
             - name: IP
-              value: ""
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.hostIP
```

Comment:
Does OpenShift define somewhere exactly what `hostIP` is?  Given your scenario with internal and external IPs, it's not obvious from the naming that it would be the right one of those.

Reply:
OpenShift use the IP address from an interface that is configured with the default route. We override the IP address with the host variable `openshift_ip`.